sql-plugin/src/main/330+/scala/com/nvidia/spark/rapids/shims/Spark33XShims.scala

```diff
@@ -312,19 +312,116 @@ trait Spark33XShims extends Spark321PlusShims with Spark320PlusNonDBShims {
         (a, conf, p, r) => new BinaryAstExprMeta[LessThanOrEqual](a, conf, p, r) {
           override def convertToGpu(lhs: Expression, rhs: Expression): GpuExpression =
             GpuLessThanOrEqual(lhs, rhs)
+        }),
+        GpuOverrides.expr[Abs](
+          "Absolute value",
+          ExprChecks.unaryProjectAndAstInputMatchesOutput(
+            TypeSig.implicitCastsAstTypes, TypeSig.gpuNumeric + TypeSig.ansiIntervals,
+            TypeSig.cpuNumeric + TypeSig.ansiIntervals),
+          (a, conf, p, r) => new UnaryAstExprMeta[Abs](a, conf, p, r) {
+            val ansiEnabled = SQLConf.get.ansiEnabled
+
+            override def tagSelfForAst(): Unit = {
+              if (ansiEnabled && GpuAnsi.needBasicOpOverflowCheck(a.dataType)) {
+                willNotWorkInAst("AST unary minus does not support ANSI mode.")
+              }
+            }
+
+            // ANSI support for ABS was added in 3.2.0 SPARK-33275
+            override def convertToGpu(child: Expression): GpuExpression = GpuAbs(child, ansiEnabled)
+          }),
+      GpuOverrides.expr[UnaryMinus](
+        "Negate a numeric value",
+        ExprChecks.unaryProjectAndAstInputMatchesOutput(
+          TypeSig.implicitCastsAstTypes,
+          TypeSig.gpuNumeric + TypeSig.ansiIntervals,
+          TypeSig.numericAndInterval),
+        (a, conf, p, r) => new UnaryAstExprMeta[UnaryMinus](a, conf, p, r) {
+          val ansiEnabled = SQLConf.get.ansiEnabled
+
+          override def tagSelfForAst(): Unit = {
+            if (ansiEnabled && GpuAnsi.needBasicOpOverflowCheck(a.dataType)) {
+              willNotWorkInAst("AST unary minus does not support ANSI mode.")
+            }
+          }
+
+          override def convertToGpu(child: Expression): GpuExpression =
+            GpuUnaryMinus(child, ansiEnabled)
+        }),
+      GpuOverrides.expr[UnaryPositive](
+        "A numeric value with a + in front of it",
+        ExprChecks.unaryProjectAndAstInputMatchesOutput(
+          TypeSig.astTypes,
+          TypeSig.gpuNumeric + TypeSig.ansiIntervals,
```

Comment:
If the only difference in the shim is the type checks, it would be nice to shim the types to pass rather than make a copy of the entire override that has to be kept in sync other than just which types to pass as parameters.  Comment applies elsewhere where we're making a copy only to update the type checks.

Reply:
Good idea!

Reply:
https://github.com/NVIDIA/spark-rapids/issues/5150